vmupdate/agent/source/dnf/dnf5_api.py

```diff
@@ -133,6 +133,49 @@ def __init__(self, weight: int, log):
         self.package_bytes = {}
         self.package_names = {}
         self.count = 0
+        self.fetching_notified = False
+
+    def add_new_download(
+            self, _user_data, description: str, total_to_download: float
+    ) -> int:
+        """
+        Notify the client that a new download has been created.
+
+        :param _user_data: User data entered together with url/package to download.
+        :param description: The message describing new download (url/packagename).
+        :param total_to_download: Total number of bytes to download.
+        :return: Associated user data for new download.
+        """
+        self.count += 1
+        self.bytes_to_fetch += total_to_download
+        self.package_bytes[self.count] = 0
+        self.package_names[self.count] = description
+        # downloading is not started yet
+        self.notify_callback(0)
+        return self.count
+
+    def progress(
+            self, user_cb_data: int, total_to_download: float, downloaded: float
+    ) -> int:
+        """
+        Download progress callback.
+
+        :param user_cb_data: Associated user data obtained from add_new_download.
+        :param total_to_download: Total number of bytes to download.
+        :param downloaded: Number of bytes downloaded.
+        """
+        if not self.fetching_notified:
+            print(f"Fetching {self.count} packages", flush=True)
```

Comment:
Maybe include total download size too here?

Reply:
Good idea, I did it for all package managers